modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/OpenAPIResourceImpl.java

```diff
@@ -53,10 +72,203 @@ public OpenAPIResourceImpl(
+
+					customParameter.name(parameter.getName());
+					customParameter.in(parameter.getIn());
+					customParameter.required(parameter.getRequired());
+					customParameter.schema(parameter.getSchema());
+
+					return customParameter;
+				}
+			).collect(
+				Collectors.toList()
+			);
+
+		String currentObjectName = StringUtil.lowerCaseFirstLetter(
+			_currentObjectDefinition.getShortName());
+
+		_customizeParameter(
+			customRelationshipPutOperationParameters, _CURRENT_OBJECT_ID,
+			currentObjectName + "Id", false);
+
+		String relatedObjectName = StringUtil.lowerCaseFirstLetter(
+			relatedObjectDefinition.getShortName());
+
+		_customizeParameter(
+			customRelationshipPutOperationParameters, _RELATED_OBJECT_ID,
+			relatedObjectName + "Id", false);
+
+		_customizeParameter(
+			customRelationshipPutOperationParameters, _RELATIONSHIP_NAME,
+			objectRelationship.getName(), true);
+
+		Operation customRelationshipPutOperation = new Operation();
+
+		customRelationshipPutOperation.tags(relationshipPutOperation.getTags());
+		customRelationshipPutOperation.operationId(
+			StringBundler.concat(
+				"put", StringUtil.upperCaseFirstLetter(currentObjectName),
+				StringUtil.upperCaseFirstLetter(objectRelationship.getName()),
+				StringUtil.upperCaseFirstLetter(relatedObjectName)));
+		customRelationshipPutOperation.parameters(
+			customRelationshipPutOperationParameters);
+		customRelationshipPutOperation.responses(
+			relationshipPutOperation.getResponses());
+
+		return new PathItem().put(customRelationshipPutOperation);
+	}
+
+	private void _customizeParameter(
+		List<Parameter> parameters, String parameterName,
+		String newParameterName, boolean shouldBeDeleted) {
+
+		Stream<Parameter> parametersStream = parameters.stream();
+
+		parametersStream.filter(
+			parameter -> parameter.getName(
+			).equals(
+				parameterName
+			)
+		).findFirst(
+		).ifPresent(
+			parameter -> {
+				if (shouldBeDeleted) {
+					parameters.remove(parameter);
+				}
+
+				parameter.setName(newParameterName);
+			}
+		);
 	}
 
+	private List<String> _filterEndpoints(
+		List<String> relationshipsEndpoints, Predicate<String> predicate) {
+
+		Stream<String> relationshipsEndpointsStream =
+			relationshipsEndpoints.stream();
+
+		return relationshipsEndpointsStream.filter(
+			predicate
+		).collect(
+			Collectors.toList()
+		);
+	}
+
+	private List<String> _getRelationshipsEndpoints(Paths paths) {
+		Set<String> keySet = paths.keySet();
+
+		Stream<String> keySetStream = keySet.stream();
+
+		return keySetStream.filter(
+			pathItem -> pathItem.contains(_RELATIONSHIP_NAME)
+		).collect(
+			Collectors.toList()
+		);
+	}
+
+	private List<String> _removeRelationshipsOpenAPIEndpoints(
+		OpenAPI openAPI, List<String> endpoints, Predicate<String> predicate) {
+
+		List<String> endpointsToRemove = _filterEndpoints(endpoints, predicate);
+
+		Paths paths = openAPI.getPaths();
+
+		endpointsToRemove.forEach(
```

Comment:
Don't know if these forEach methods give us some advantage compared with a for each loop that may be more clear. Let's see what Brian thinks